MapboxCoreNavigation/RouteController.swift

```diff
@@ -472,3 +618,134 @@ extension RouteController: CLLocationManagerDelegate {
         incrementRouteProgress(alertLevel, location: location, updateStepIndex: updateStepIndex)
     }
 }
+
+struct SessionState {
+    let identifier = UUID()
+    var departureTimestamp: Date?
+    var arrivalTimestamp: Date?
+    
+    var totalDistanceCompleted: CLLocationDistance = 0
+    
+    var numberOfReroutes = 0
+    var lastReroute: Date?
+    
+    var currentRoute: Route!
+    var currentRequestIdentifier: String?
+    
+    var originalRoute: Route!
```

Comment:
Implicitly unwrapped optionals are unsafe. Make this property and `currentRoute` non-optional and pass them in when initializing the SessionState below.